clients/java/src/main/java/io/camunda/client/impl/worker/JobStreamerImpl.java

```diff
@@ -188,6 +196,53 @@ private void lockedOpen() {
     control.whenCompleteAsync((ignored, error) -> handleStreamComplete(error), executor);
     streamControl = control;
     LOGGER.debug("Opened job stream of type '{}' for worker '{}'", jobType, workerName);
+
+    if (streamTimeout != null) {
+      LOGGER.debug(
+          "Scheduling recreation of the job stream of type '{}' for worker '{}' after '{}s'",
+          jobType,
+          workerName,
+          streamTimeout.getSeconds());
+      if (scheduledRecreationTriggerTask != null) {
+        scheduledRecreationTriggerTask.cancel(true);
```

Comment:
🔧 I'm pretty sure this is idempotent, but double check, or add a check to cancel only if not cancelled.